### DISCUSSION TITLE
Which ChatGPT alternatives are you using?

### DISCUSSION THREAD

**Discussion by JosephTLyons** [zed-industries/zed#6675] [Feature Requests]
For those of you using something outside of [ChatGPT](https://chat.openai.com), what are you using and why?  Please search for and upvote existing replies.

**Comment by widersky** [comment]
JetBrains LLM (in closed-beta) is unbeatable for me in case of accuracy and actual help in even more complex problems. CoPilot gets lost to a bit more complicated issues (for example, it can reverse the logic of the code), and ChatGPT itself is even worse 🤷‍♂️

**Reply by BKG01** [reply]
Any way to sign up for access?

**Comment by TerminalFi** [comment]
Honestly, this should live in a plugin. Just sounds like feature bloat

**Comment by enzime4u** [comment]
Using Codeium , it amazing and also FREE ! Does an amazing job

**Comment by gldtn** [comment]
Maybe Google Bard is a good option here, although it's still behind chatGPT. Also I believe their API is still in beta and you have to get on a waiting list.

**Comment by nym21** [comment]
Any from this list: https://ollama.ai/library, just being compatible with ollama's server would be awesome

**Reply by YurySolovyov** [reply]
Which in turn uses/wraps `llama.cpp` server, but both are great projects ✨

**Reply by chuikoffru** [reply]
It would be great to use codellama (ollama) as AI assist instead the Copilot in Zed.

**Comment by damywise** [comment]
Idk the context but I assume you mean in general?

1. Phind (vscode extension and web) for mostly programming, on vscode I use it along with Codeium.
Codeium has auto complete but phind has better capability.
2. Perplexity for more broad general search. If I can't find something with Phind, sometimes I can find it in Perplexity, and vice versa.

I use them interchangibly.

**Comment by LucasJohnNyamhanga** [comment]
Am on GitHub Copilot.

**Comment by Pipboyguy** [comment]
Mistral Medium is fantastic

**Comment by roninreilly** [comment]
https://openrouter.ai

**Reply by codegod100** [reply]
zed needs to add this. you can access most other ai from here.

**Comment by rohanpoudel2** [comment]
I am using Phind

**Comment by mindaslab** [comment]
https://tabby.tabbyml.com/ I am using it in VSCodium. This is also available as language server, so may be it might be easy to integrate into Zed.

**Comment by SMak06** [comment]
Can I use any of the alternative in Zed instead of chatGPT?

If so, how do I remove the existing API key that I entered (it's not working for me)?

**Reply by Moshyfawn** [reply]
You have already asked how to reset the API key three times, and one of those threads already [contains the answer](https://github.com/zed-industries/zed/discussions/6650#discussioncomment-7488716) to your other question. Please refrain from posting the same question multiple times in different threads.

**Reply by OscarJPrice** [reply]
You can use Github copilot, but that appears to be it.

**Comment by simonmcconnell** [comment]
Cody by Sourcegraph, because it is the first and only one I've tried.

**Reply by tumma72** [reply]
I agree Cody in VS Code is at the moment the superior choice. The ability to switch model and the ability to keep context of the local code in a vectorized and embedded format make it an extremely valuable asset. Additionally the freedom to switch model and even use local Ollama installed modem when offline is great.

**Comment by joshrutkowski** [comment]
[CodeWhisperer](https://aws.amazon.com/codewhisperer)

**Reply by Moshyfawn** [reply]
Have you been using CW since Fig or just CW?

**Reply by joshrutkowski** [reply]
I’ll clarify CW (as in rebranded Fig), yes I use that. But primarily I meant the IntelliJ and VSCode extensions.

**Reply by KoStard** [reply]
+1 of codewhisperer, would be happy to see support for it in Zed

**Comment by Frank-vdm** [comment]
Im using a combination of copilot chat, openIA and a few others, in usually it depends on my project, ideally i would like to be able to configure which assistant to use in the assistant panel.

that being said since we already have github copilot intergration could we expect github copilot chat to become available?

**Comment by AugustinMauroy** [comment]
I'm using [mistral](https://mistral.ai)
it's open source

**Comment by phtn** [comment]
please add [phind](https://phind.com)

**Comment by clearfram3** [comment]
I've used Cody from SourceGraph, Double.bot, and GPT4 in chatGPT. Mistral's Medium, 8x7, and Large are really good, and Claude 3 looks amazing.

**Reply by jackbravo** [reply]
+1 to double.bot and Claude 3

**Reply by ggoonnzzaallo** [reply]
I'm one of the co-founders of [double.bot,](https://double.bot) appreciate the shoutout! If there's anything we can do to improve the product, let me know :)

**Reply by tendiesfortwo** [reply]
+1 for double.bot!

**Comment by adamierymenko** [comment]
I've had decent results from DeepSeek Coder and CodeLlama models running in lm-studio locally using plugins in VSCode like Continue. LM-Studio is a local GUI for running local models that can start its own local OpenAI-compatible API server. (Bonus: running large local LLMs is one of the only things I can do on my M1 Max to make the CPU fans spin like an old Intel MacBook 😺 )

I'd *really* like to see support for configuring your own AI endpoints and using things like lm-studio and ollama.

Claude 3 is also excellent and if/when it's possible to integrate I'd love to see support for that too.

Is there a way to sponsor features?

**Comment by tendiesfortwo** [comment]
First step is ditching the OpenAI models (at least until they release something better). Claude 3 Opus is clearly superior.

Within the tools available right now that use Claude 3 (not many tbh), my favorite so far has been [double.bot ](https://double.bot). Super generous free trial too to get a feel for it

**Comment by logiota** [comment]
Ollama ftw!

**Comment by phaynes** [comment]
In addition to chatGPT, I use Gemini and Claude. I think the Groq is going to be a transformational technology. 

As I am a software architect, I do more text editing in markdown than most.  The different engines allow me to review documents t to a higher standard than another by it self.

**Comment by enzime4u** [comment]
which ones are supported by Zed ? 
Also is there a new guide to setup ChatGPT on Zed ? since it's a pain to set it up

**Comment by jenil777007** [comment]
ChatGPT & Claude 💯

**Comment by tokiory** [comment]
Definitely Codeium.

**Comment by ggoonnzzaallo** [comment]
Anyone using VS Code as their main IDE should check out [Double.bot](https://double.bot)
